pkg/apis/logging/v1/cluster_log_forwarder_types.go

```diff
@@ -0,0 +1,76 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// ClusterLogForwarder is the schema for the `clusterlogforwarder` API.
+//
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:scope=cluster
+// +kubebuilder:subresource:status
+type ClusterLogForwarder struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ClusterLogForwarderSpec   `json:"spec,omitempty"`
+	Status ClusterLogForwarderStatus `json:"status,omitempty"`
+}
+
+// ClusterLogForwarderSpec defines the desired state of ClusterLogForwarder
+type ClusterLogForwarderSpec struct {
+	// Outputs are named output destinations for log messages.
+	//
+	// +optional
```

Comment:
from an implementation perspective, what does having a `pipeline` defined without `outputs` net users?

Reply:
Nothing, will make that required non-empty